uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/OneAuthTestApp.java

```diff
@@ -0,0 +1,200 @@
+import com.microsoft.identity.client.ui.automation.interaction.microsoftsts.AadPromptHandler;
+import com.microsoft.identity.client.ui.automation.logging.Logger;
+import com.microsoft.identity.client.ui.automation.utils.CommonUtils;
+import com.microsoft.identity.client.ui.automation.utils.UiAutomatorUtils;
+
+import org.junit.Assert;
+
+import java.util.ArrayList;
+import java.util.List;
+
+// https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2516682
+
+/**
+ * Infrastructure for testing OneAuthTest app
+ */
+public class OneAuthTestApp extends App implements IFirstPartyApp {
+    private final static String TAG = "OneAuthTestApp";
+    private final static String ONEAUTH_TESTAPP_PACKAGE_NAME = "com.microsoft.oneauth.testapp";
+    private final static String ONEAUTH_TESTAPP_NAME = "OneAuth Testapp";
+    private final static String ONEAUTH_TESTAPP_APK = "OneAuth.apk";
+
+    public OneAuthTestApp() {
+        super(ONEAUTH_TESTAPP_PACKAGE_NAME, ONEAUTH_TESTAPP_NAME, new LocalApkInstaller());
+    }
+
+    public OneAuthTestApp(@NonNull final IAppInstaller appInstaller) {
+        super(ONEAUTH_TESTAPP_PACKAGE_NAME, ONEAUTH_TESTAPP_NAME, appInstaller);
+        localApkFileName = ONEAUTH_TESTAPP_APK;
+    }
+
+    @Override
+    protected void initialiseAppImpl() {
+        // nothing needed here
+    }
+
+    @Override
+    public void handleFirstRun() {
+        CommonUtils.grantPackagePermission();
+    }
+
+    @Override
+    public void addFirstAccount(@NonNull final String username,
+                                @NonNull final String password,
+                                @NonNull final FirstPartyAppPromptHandlerParameters promptHandlerParameters) {
+        Logger.i(TAG, "Adding First Account..");
+        // sign in with supplied username/password
+        signIn(username, password, promptHandlerParameters);
+    }
+
+    @Override
+    public void addAnotherAccount(@NonNull final String username,
+                                  @NonNull final String password,
+                                  @NonNull final FirstPartyAppPromptHandlerParameters promptHandlerParameters) {
+        // nothing needed here
+    }
+
+    public String acquireTokenInteractive(@NonNull final String username,
+                                          @NonNull final String password,
+                                          @NonNull final FirstPartyAppPromptHandlerParameters promptHandlerParameters) {
+        signIn(username, password, promptHandlerParameters);
+        return getTokeSecret();
+    }
+
+    public String acquireTokenSilent() {
+        // Click Get Access token button
+        UiAutomatorUtils.handleButtonClick("com.microsoft.oneauth.testapp:id/get_access_token_button");
+        try {
+            // Add a delay so that UI is updated with the token successfully
+            Thread.sleep(2000);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
```

Comment:
How about using a logger to print the stacktrace?

Reply:
Its following existing pattern from the codebase. I found occurences of thread sleeping in code, or catching interrupted exceptions, and using same e.printStackTrace.